lib/src/cli.rs

```diff
@@ -610,6 +619,47 @@ async fn usroverlay() -> Result<()> {
         .into());
 }
 
+/// Implementation of `bootc build commit`
+/// async fn lint() -> Result<()> {
+#[context("linting")]
+fn lint() -> Result<()> {
+    if !ostree_ext::container_utils::is_ostree_container()? {
+        anyhow::bail!("Not in a ostree container, this command only verifies ostree containers.");
+    }
+
+    // check for the existance of the /var/run directory
+    // if it exists we need to check that it links to /run if not error
+    // if it does not exist error.
+
+    let root = cap_std::fs::Dir::open_ambient_dir("/", cap_std::ambient_authority())?;
+    let varRunLint = find_var_run_symlink_fs(&root)?;
```

Comment:
Rust naming is snake_case not camelCase.  But in this case we're not doing anything with the return value, so we can just drop it.

Let's have this explicitly run each lint as its own function from a standard array, something like:

```
   let lints = [check_var_run, check_kernel];
   for lint in lints {
     lint()?;
   }
```

Specifically I think we can simplify `check_var_run()` to something like:

```
fn check_var_run(root: &Dir) -> Result<()> {
   if let Some(meta) = root.symlink_metadata_optional("var/run")? {
       if !meta.is_symlink()) {
          anyhow::bail!("Not a symlink: var/run");
       }
   }
   Ok(())
}
```

for example?

Reply:
On the lint execution side, see https://github.com/coreos/rpm-ostree/blob/4f6dfd089c3686bbe1c9160332ce5758524c5865/rust/src/composepost.rs#L426 which is a *very* similar pattern of calling multiple functions.

Reply:
Ah, ok I think we have done this now. Thank you; it looks much better.